src/plugins/ec.ts

```diff
@@ -172,6 +173,33 @@ export class ECPlugin extends BasePlugin {
             }, {});
     }
 
+    private convertProductPriceToNumber(product: Product) {
+        if (product.price) {
+            product.price = this.tryConvertStringPriceToNumber(product.price);
+        }
+
+        return product;
+    }
+
+    private convertImpressionPriceToNumber(impression: BaseImpression) {
+        if (impression.price) {
+            impression.price = this.tryConvertStringPriceToNumber(impression.price);
+        }
+
+        return impression;
+    }
+
+    private tryConvertStringPriceToNumber(price: Number | string): Number | string {
```

Comment:
I think we can leave these as `number | string` because when we create a Number, we don't use the `new` keyword. Hence, it creates a primitive number which is compatible with `number`
<img width="300" alt="image" src="https://user-images.githubusercontent.com/8978908/204050867-69616f1f-bbe6-420e-917a-1749c6bd84a7.png">

```suggestion
    private tryConvertStringPriceToNumber(price: number | string): number | string {
```

This means we don't need to change `ProductProperties`'s `price` to a `Number | string`, but can put it as `number | string`. It should be more consistent with how the other properties that are `number`s are defined.

Reply:
I agree with mister Nick